.github/workflows/ci.yml

```diff
@@ -334,3 +332,20 @@ jobs:
       - name: Push load test result
         if:  ${{ success() && github.ref_name == 'v3'}}
         run: git push 'https://github-actions:${{ secrets.GITHUB_TOKEN }}@github.com/nginx/agent.git' benchmark-results:benchmark-results
+  publish-packages:
+    name: Publish NGINX Agent v3 packages
+    if: ${{ github.event_name =='pull_request' &&
+            github.base_ref == 'v3' &&
+            github.event.pull_request.merged == true &&
+            !github.event.pull_request.head.repo.fork }}
+    needs: [ lint, unit-test, performance-tests,
+             load-tests, official-oss-image-integration-tests,
+             official-plus-image-integration-tests,
+             race-condition-test ]
+    uses: ./.github/workflows/release-branch.yml
+    with:
+      packageVersion: "3.0.0"
+      packageBuildNo: "${{ github.run_number }}"
+      uploadAzure: true
+      publishPackages: true
+      releaseBranch: "v3"
```

Comment:
I think you also need to add the following to this job so that it has access to the secrets and has the required permissions
```
    secrets: inherit
    permissions:
      contents: read
      id-token: write
```

Reply:
Updated, initial thought was we also add permissions explicitly to the jobs which need them like [upload_packages](https://github.com/nginx/agent/blob/publish_package/.github/workflows/release-branch.yml#L223)